overlord/state/taskrunner.go

```diff
@@ -336,20 +336,22 @@ func (r *TaskRunner) Ensure() {
 			}
 			continue
 		}
+
+		if mustWait(t) {
+			// Dependencies still unhandled.
+			continue
+		}
+
 		if status == UndoStatus && handlers.undo == nil {
-			// Cannot undo. Revert to done status.
```

Comment:
we might want to keep this bit but below the new comment addition

Reply:
Ok. done.